### PR TITLE
Add rmrf shorthand for rm(x, force=true, recursive=true)

### PR DESCRIPTION
**Pull request by ararslan**
As suggested by Stefan in https://github.com/JuliaLang/julia/pull/32335#issuecomment-519661907.

### CODE REVIEW

File: base/file.jl [hunk 1]
```diff
@@ -280,6 +281,18 @@ function rm(path::AbstractString; force::Bool=false, recursive::Bool=false)
     end
 end
 
+"""
+    rmrf(path::AbstractString)
+
+Forcibly and recursively remove the provided path.
+Equivalent to `rm(path, force=true, recursive=true)`.
```
Comment (StefanKarpinski) [inline base/file.jl]:
```suggestion
Shorthand for `rm(path, force=true, recursive=true)`.
```

**Review by StefanKarpinski — APPROVED** [review]
That's a nice tidy PR!

File: base/file.jl (hunk 1 above)
Comment (StefanKarpinski) [reply base/file.jl]: My reasoning being that we often say "equivalent to" when we mean "does effectively the same thing but in a different way" whereas here it's literally the same as that.

**Comment by ararslan** [comment]
Wow, it's been a while since I've seen a straight flush of green on CI.

File: base/file.jl [hunk 2]
```diff
@@ -280,6 +281,18 @@ function rm(path::AbstractString; force::Bool=false, recursive::Bool=false)
...
+
+Forcibly and recursively remove the provided path.
+Shorthand for `rm(path, force=true, recursive=true)`.
+
+See also [`rm`](@ref).
+
+!!! compat "Julia 1.3"
+    This function requires Julia 1.3 or later.
+"""
+rmrf(path::AbstractString) = rm(path, force=true, recursive=true)
```
Comment (fredrikekre) [inline base/file.jl]:
```suggestion
rmrf(path) = rm(path, force=true, recursive=true)
```

Comment (StefanKarpinski) [reply base/file.jl]: Agree. No reason to constrain this.

Comment (ararslan) [reply base/file.jl]: Just did it originally for consistency with `rm`.

**Comment by ararslan** [comment]
Uh, all the CI processes got `-9` killed.

**Comment by StefanKarpinski** [comment]
The system AI detected that you were trying to `rmrf` things and got angry.

**Comment by ararslan** [comment]
Elliot was doing a bit of buildbot surgery which is why the builds got rekt. Closed and reopened to rerun.

**Comment by ararslan** [comment]
All good?

**Comment by rfourquet** [comment]
Is the only motivation for use in `mktempdir`?

**Comment by StefanKarpinski** [comment]
I often want to `rmrf` something and writing out `rm(path, recursive=true, force=true)` is a bit of an awkward fistful to type. Having `rmrf` as a shortcut would be very handy, imo.

**Comment by rfourquet** [comment]
I admit that i personally use (in bash) `rm -rf` and plain `rm` more often than with other options... but is it clear we won't start adding `rmr`, `rmf`, and aliases for other frequent functions with long kwargs?

**Comment by StefanKarpinski** [comment]
The intermediate cases don't have so much motivation: writing `rm(path, force=true)` and `rm(path, recursive=true)` are both fine, imo. It's the much more common and annoyingly long `rm(path, recursive=true, force=true)` that needs a shortcut.

**Comment by ararslan** [comment]
NEWS conflict resolved.

**Comment by stevengj** [comment]
`rmrf` seems pretty obscure to me.  Especially since this is a dangerous operation, I prefer the more verbose `rm(x, force=true, recursive=true)` … is it really so common that we need an abbreviation?

**Comment by ararslan** [comment]
I think the motivation was stronger when https://github.com/JuliaLang/julia/pull/32335 was under consideration.

**Comment by ararslan** [comment]
Triage: Nah

**Comment by StefanKarpinski** [comment]
I think the triage decision was about whether to merge for 1.3, we might still decide to do it later. But the PR could always be reopened.

**Comment by ararslan** [comment]
> I think the triage decision was about whether to merge for 1.3

Oh, that is not what it sounded like to me; it sounded like there is insufficient motivation for it in general, and there are safety concerns as Steven noted.

**Comment by StefanKarpinski** [comment]
FWIW, there are 32 places where this function would be called in Base + stdlibs.

**Pull request closed without merging**